pkg/api/client.go

```diff
@@ -38,6 +38,15 @@ type ClientOptions struct {
 	// Default is no logging.
 	Log io.Writer
 
+	// LogIgnoreEnv disables respecting the GH_DEBUG environment variable for logging.
+	LogIgnoreEnv bool
```

Comment:
After thinking about this more, I am not sure that this option is necessary. I understand its purpose, but can't think of a good use case where we want to allow an extension author to ignore `GH_DEBUG`. 

Additionally, I think our users would find the behavior of this option a bit confusing without further comments describing its behavior.

Reply:
> After thinking about this more, I am not sure that this option is necessary. I understand its purpose, but can't think of a good use case where we want to allow an extension author to ignore `GH_DEBUG`.

Thanks for chiming in. One consumer of go-gh that wants to use this option is the CLI itself, since GitHub CLI already parses environment variables itself and wants to prevent underlying libraries to parse the same environment variables, lest some inconsistencies or conflicts arise.

Additionally, an extension author might want to disable env-controlled logging if it already offers the user granular controls for logging.

As a general rule of thumb, if a generic Go library automatically respects environment variables, it's a good idea to offer a way for the consumer of said library to skip respecting environment variables in cases where that is not desired. At the end of the day, environment variables are essentially global variables shared across modules and even processes, and to mitigate their potential harms, developers should have ways to put up boundaries to limit the propagation (or effect) of these variables.

Reply:
That all makes sense to me, my only thought was that if an extension author wanted to ignore `GH_DEBUG` they could provide a value for the `Log` option making `LogIgnoreEnv` feel a bit redundant.